### ISSUE TITLE
Czech words defined in dictionaries are not skipped during spellcheck process

### ISSUE THREAD

**Issue by PatrickHammond66** [facelessuser/pyspelling#134]
We have defined a non-English dictionary to skip words that appear in our documentation that are not in English from being reported as spelling errors. 

However, it appears that there are certain characters which Aspell does not recognise when defined in these dictionaries. 

The following words appear in our documentation, and are not skipped when defined in a dictionary: 

Matična
Reģistrācijas
Įmonės
številka
Daňové
identifikační
číslo

Thus we can guess it's something in the encoding of these characters: č, ā, Į, š, ň

**Comment by facelessuser** [comment]
It might be a limitation of the dictionary. I imagine spellchecking works with a Czech dictionary, so I imagine there is something specifically incompatible with the dictionary.

So could one theoretically build a dictionary compatible with English and Czech and then add ignore lists to that? It seems like it may be possible.

1. Need to confirm these work fin in a Czech dictionary (sanity check, I imagine it does work).
2. Need to see if one can provide multiple dictionaries when spellchecking (maybe both Czech and English). And if so, see if this works with PySpelling, and if not see if we can add support.

If 2 is not possible, maybe we can add a filter that can target certain words with regex or something.

Most likely it is not a "bug" on our part, but maybe we can add an enhancement to better deal with such cases.

@gir-bot remove S: triage
@gir-bot add T: enhancement

**Comment by PatrickHammond66** [comment]
Regex filter for unwanted characters might be best approach indeed. I am a bit worried about introducing more than English dictionary in case it increases false positives (misspellings in EN that are correct in CS), but might be overestimating the chances there. I did try to provide two "lang:" parameters under aspell in the matrix, but it caused problems.

**Comment by facelessuser** [comment]
@PatrickHammond66, you should be able to work around this with this:

```yml
  - pyspelling.filters.context:
      context_visible_first: true
      delimiters:
      - open: '\b\w*?[čāĮšň]'
        close: '\b'
```

It just dawned on me that this is already possible

**Comment by facelessuser** [comment]
Originally, the context filter was added for adding start and end markers, but the way that the one works above, the start and end should encompass the entire word.

**Comment by PatrickHammond66** [comment]
This worked! (I should have thought about this as I've added regexs) At least we know of the issue now too for future users!

**Comment by facelessuser** [comment]
Yeah, the solution is a little opaque, but I think it is a reasonable solution. Before I close this issue though, I will confirm that such words can be excluded with Czech dictionaries. I want to make sure this is nothing on our side (PySpelling) that makes using these characters difficult.

**Comment by facelessuser** [comment]
So I ran things through a cs dictionary, and I had some issues until I ran the content through the text filter and normalized Unicode to NFD. This is more informational. While this helps in English languages, it still isn't enough, but it is sufficient in CS dictionaries.

```yaml
  pipeline:
  - pyspelling.filters.text:
      normalize: 'NFD'
```

So, a couple of things can be noted:

1. Dictionaries help determine what is accepted. English dictionaries won't accept certain characters even if you normalize the Unicode.

2. International languages that are using special characters may not match correctly without Unicode normalization.

In your case, I still think using regex is best as the English dictionary, even if you normalize the Unicode will still not catch all Czech words. If you were doing Czech content through a cs dictionary, then I may suggest just normalizing the content, but since you are not, regex is best.

This has been interesting as I obviously approach things through an English lens, so seeing some actual cases of international usage helps me to gain a better understanding.
